### PR TITLE
[Feautre] Beneficiary designation

### LINKED ISSUE #123: [Feature] [Enrollment] Beneficiary designation
Implement sketch present at https://www.sketch.com/s/09cd8050-fd53-4c45-a4f3-ccdef3efeae7/a/GVzJL7

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #123 

close with associatedemployers/igneous#358

image

image

### CODE REVIEW

File: app/controllers/account/benefits/enrollment/beneficiaries.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  @action
+  async saveLife () {
+    if (this.model.contingentLife.dirtyType === 'created') {
+      this.model.enrollment.beneficiaries.addObject(this.model.contingentLife);
+    }
+
+    if (this.model.primaryLife.dirtyType === 'created') {
+      this.model.enrollment.beneficiaries.addObject(this.model.primaryLife);
+    }
+  }
```
Comment (James1x0) [inline app/controllers/account/benefits/enrollment/beneficiaries.js:17]: This is going to duplicate beneficiaries

**Review by James1x0 — CHANGES_REQUESTED** [review]

File: app/config/statics.js [hunk 2]
```diff
@@ -300,6 +300,14 @@ const employeeMutableAttrs = [
   'phone'
 ];
 
+const dependents = [
+  'Daughter',
+  'Son',
+  'Spouse',
+  'Court Appointed Dependent',
+  'Domestic Partner'
+];
```
[the remark this answers is not in the document]
Comment (JavaLavaMT) [reply app/config/statics.js:309]: 🏁

File: app/models/beneficiary.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  ssn:          attr('string'),
+  relationship: attr('string'),
+  dateOfBirth:  attr('date'),
+  percentage:   attr('number'),
+  created:            attr('date', {
+    defaultValue () {
+      return new Date();
+    }
+  })
+});
```
[the remark this answers is not in the document]
Comment (JavaLavaMT) [reply app/models/beneficiary.js:19]: 🏁

File: app/controllers/account/benefits/enrollment/beneficiaries.js (hunk 1 above)
Comment (JavaLavaMT) [reply app/controllers/account/benefits/enrollment/beneficiaries.js:17]: I was way over thinking this

Comment (JavaLavaMT) [reply app/controllers/account/benefits/enrollment/beneficiaries.js:17]: 🏁

**Pull request merged**